### DISCUSSION TITLE
AssertionError : assert (force is not None) and len(coord) > 0 and len(cell) > 0

### DISCUSSION THREAD

**Discussion by tirtha-v** [deepmodeling/dpdata#473] [Q&A] [answered]
In reference to earlier discussions on Assertion error:

Forces missing: [link1](https://github.com/deepmodeling/dpdata/discussions/285)
Section of information of POTCAR is missing: [link2](https://github.com/deepmodeling/dpdata/discussions/434) 

The OUTCAR file I am using consists of forces for all iterations along with the information from POTCAR file. In spite of this, I am still facing assertion error while using dpdata (data preparation for Deepmd).
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/anaconda3/envs/deepmd/lib/python3.9/site-packages/dpdata/system.py", line 281, in __init__
    self.from_fmt(
  File "/home/anaconda3/envs/deepmd/lib/python3.9/site-packages/dpdata/system.py", line 318, in from_fmt
    return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)
  File "/home/anaconda3/envs/deepmd/lib/python3.9/site-packages/dpdata/system.py", line 1162, in from_fmt_obj
    data = fmtobj.from_labeled_system(file_name, **kwargs)
  File "/home/anaconda3/envs/deepmd/lib/python3.9/site-packages/dpdata/plugins/vasp.py", line 79, in from_labeled_system
    ) = dpdata.vasp.outcar.get_frames(
  File "/home/anaconda3/envs/deepmd/lib/python3.9/site-packages/dpdata/vasp/outcar.py", line 76, in get_frames
    coord, cell, energy, force, virial, is_converge = analyze_block(
  File "/home/anaconda3/envs/deepmd/lib/python3.9/site-packages/dpdata/vasp/outcar.py", line 145, in analyze_block
    assert (force is not None) and len(coord) > 0 and len(cell) > 0
AssertionError
```
The file size being big has been initially split and then zipped. 
[out1.zip](https://github.com/deepmodeling/dpdata/files/11516783/out1.zip)
[out2.zip](https://github.com/deepmodeling/dpdata/files/11516786/out2.zip)
[out3.zip](https://github.com/deepmodeling/dpdata/files/11516789/out3.zip)

**Comment by wanghan-iapcm** [comment]
Hello I find you are using vasp 4.6.x, the order of the vasp sections of coordinates/forces and energy appears in the outcar is different from vasp 5.x. You are welcome to contribute the support of the outcar vasp 4.6.x.

**Comment by tirtha-v** [comment, marked as answer]
I realised that if the energy token string is modified, it gives the desirable results.
I have changed the energy token string in outcar.py file according to the information present in the OUTCAR file.

**Reply by wanghan-iapcm** [reply]
do you mean the token used by vasp 4.6 is different from more resent version?  
could you please report what were the original and changed energy tokens?

**Reply by tirtha-v** [reply]
The original energy token provided in the package is "free  energy   TOTEN" which indicates the energy value after every SCF step. I replaced the energy token with "% ion-electron   TOTEN" which selects energy after an iteration is over.
